### PR TITLE
Support for Item categorization in Jenkins 2

### PR DESCRIPTION
**Pull request by recena**
This PR belongs to this [JENKINS-31162](https://issues.jenkins-ci.org/browse/JENKINS-31162) issue and allows to categorize the Matrix projects in a right way.

You can find another [example](https://github.com/jenkinsci/maven-plugin/pull/62) in Maven Project Plugin.

@reviewbybees /cc @daniel-beck

### CODE REVIEW

File: src/main/java/hudson/matrix/MatrixProject.java [hunk 1]
```diff
@@ -991,10 +991,44 @@ public DescriptorImpl getDescriptor() {
     public static final DescriptorImpl DESCRIPTOR = new DescriptorImpl();
 
     public static final class DescriptorImpl extends AbstractProjectDescriptor {
+
         public String getDisplayName() {
             return Messages.MatrixProject_DisplayName();
         }
 
+        /**
+         * Needed if it wants Maven projects are categorized in Jenkins 2.x.
```
Comment (olivergondza) [inline src/main/java/hudson/matrix/MatrixProject.java]: s/Maven/Matrix/

I do not think the javadoc bares much value + there seem to be some spelling errors IFAICT. `// After Jenkins 2.0 this should be an @Override` is more than sufficient.

Comment (olivergondza) [inline src/main/java/hudson/matrix/MatrixProject.java]: Ditto for the next two methods

**Comment by recena** [comment]
@olivergondza Thanks for your feedback. Are you fine with the changes?

File: src/main/resources/hudson/matrix/Messages_zh_TW.properties [hunk 2]
```diff
@@ -20,20 +20,21 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
-MatrixProject.DisplayName=\u5efa\u7acb\u591a\u91cd\u8a2d\u5b9a\u5c08\u6848
-MatrixProject.DuplicateAxisName=\u8ef8\u7dda\u540d\u7a31\u91cd\u8907
+MatrixProject.DisplayName=\u5EFA\u7ACB\u591A\u91CD\u8A2D\u5B9A\u5C08\u6848
+MatrixProject.DuplicateAxisName=\u8EF8\u7DDA\u540D\u7A31\u91CD\u8907
```
Comment (jglick) [inline src/main/resources/hudson/matrix/Messages_zh_TW.properties]: Oof, it is impossible to see here whether you are accidentally breaking some unrelated keys. Any way to revert the unrelated changes?

**Comment by olivergondza** [comment]
My point was none of the info really belongs into javadoc (as opposed to regular comment). Anyway, :+1:

File: src/main/resources/hudson/matrix/Messages_zh_TW.properties (hunk 2 above)
Comment (recena) [inline src/main/resources/hudson/matrix/Messages_zh_TW.properties]: @jglick I would wish to avoid this. It is being very humdrum

File: src/main/java/hudson/matrix/MatrixProject.java [hunk 3]
```diff
@@ -991,10 +991,44 @@ public DescriptorImpl getDescriptor() {
...
 
+        /**
+         * Needed if it wants Matrix projects are categorized in Jenkins 2.x.
+         *
+         * TODO: After Jenkins 2.0 this should be an @Override
+         *
+         * @return A string it represents a ItemCategory identifier.
+         */
+        public String getCategoryId() {
+            return "itemcategory-standaloneprojects";
```
Comment (jglick) [inline src/main/java/hudson/matrix/MatrixProject.java]: Is this not now simply `standaloneprojects`?

Comment (recena) [inline src/main/java/hudson/matrix/MatrixProject.java]: @jglick This was like this until someone decided changing it ;). Done.

**Comment by jglick** [comment]
:bee:

**Comment by amuniz** [comment]
:bee:

**Comment by amuniz** [comment]
This is kind of virtual-dynamically-linked downstream of https://github.com/jenkinsci/jenkins/pull/2117 :smiley_cat: Just noting it here for archaeological purposes.

**Comment by recena** [comment]
@reviewbybees done

**Comment by recena** [comment]
@olivergondza Do you agree if we merge this PR?

**Comment by olivergondza** [comment]
@recena, sure, go ahead.

**Comment by recena** [comment]
@olivergondza Thanks for your time.
